app/services/courses_query.rb

```diff
@@ -49,6 +50,22 @@ def visa_sponsorship_scope
       )
   end
 
+  def study_modes_scope
+    return @scope if params[:study_types].blank?
+
+    study_modes = params[:study_types]
+    @applied_scopes[:study_modes] = study_modes
+
+    case study_modes
```

Comment:
I don't think this variable extraction is valuable enough to be worth the additional line.

Not a blocker though.

```suggestion
    @applied_scopes[:study_modes] = params[:study_types]

    case params[:study_types]
```

Reply:
Done